### PR TITLE
feat: Add more unit tests to improve coverage

### LINKED ISSUE #298: Improve code coverage (lines 90% / branches 80%)
According to the Gold badge requirements, code coverage should be
- 90% for lines
- 80% for branches

We need to increase the coverage till the required values.

### PR DESCRIPTION
**Pull request by o-kopysov**
# Description

To improve the code coverage to 90%, we have to add more unit tests.

Fixes #298 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Code cleanup/refactoring
- [ ] Documentation update
- [ ] This change requires a documentation update
- [ ] CI system update
- [X] Test Coverage update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**
- Coverage: 88.784%
- Branches: 68.533%

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**
- Coverage: 90.264%
- Branches: 70.270%

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**
- Coverage: 90.507%
- Branches: 72.201%

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**
- Coverage: 91.478%
- Branches: 77.992%

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**
- Coverage: 92.284%
- Branches: 78.958%

**Comment by o-kopysov** [comment]
The goal is achieved.

**JaCoCo Test Coverage Summary**
- Coverage: 92.855%
- Branches: 80.156%

**Review by o-konoval — APPROVED** [review]
Good job! Approved!

File: src/test/java/com/lpvs/config/SecurityConfigTest.java [hunk 1]
```diff
@@ -25,8 +38,48 @@ class SecurityConfigTest {
...
         assertNotNull(securityConfig);
     }
+
+    @Test
+    public void testOnAuthenticationSuccess() throws IOException, ServletException {
+        AuthenticationSuccessHandler successHandler = mock(AuthenticationSuccessHandler.class);
+
+        // Configure the behavior of onAuthenticationSuccess
+        doAnswer(
+                        invocation -> {
```
Comment (m-rudyk) [inline src/test/java/com/lpvs/config/SecurityConfigTest.java:52]: To much indentation

Comment (o-kopysov) [reply src/test/java/com/lpvs/config/SecurityConfigTest.java:52]: Approved by checker.

**Review by tdrozdovsky — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -23,6 +23,13 @@ jobs:
       with:
         egress-policy: audit
 
+    - name: Install Python
+      uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236
+      with:
+        python-version: '3.10'
+    - name: Install scanoss-py
+      run: pip3 install scanoss
+
```
Comment (tiokim) [inline .github/workflows/build.yml:32]: I don't think installing python and scanoss-py is necessary on the build workflow.

File: .github/workflows/findbugs.yml [hunk 3]
```diff
@@ -20,6 +20,13 @@ jobs:
       with:
         egress-policy: audit
 
+    - name: Install Python
+      uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236
+      with:
+        python-version: '3.10'
+    - name: Install scanoss-py
+      run: pip3 install scanoss
+
```
Comment (tiokim) [inline .github/workflows/findbugs.yml:29]: Is installing python and scanoss-py necessary to the findbugs workflow?

File: src/main/java/com/lpvs/service/scanner/scanoss/LPVSScanossDetectService.java [hunk 4]
```diff
@@ -69,15 +79,17 @@ public void runScan(LPVSQueue webhookConfig, String path) throws Exception {
...
-            processBuilder =
-                    new ProcessBuilder(
-                            "scanoss-py",
-                            "scan",
-                            debug ? "-t" : "-q",
-                            "--no-wfp-output",
-                            "-o",
-                            getScanResultsJsonFilePath(webhookConfig),
-                            path);
+            if (processBuilder == null) {
```
Comment (tiokim) [inline src/main/java/com/lpvs/service/scanner/scanoss/LPVSScanossDetectService.java:82]: I'm not sure if this logic is appropriate. What if processBuilder is not null along with the other pervious webhookConfig information?

**Review by tiokim — COMMENTED** [review]
Thank you very much for your great effort to improve UT coverage!
Take a look at my comments.

File: src/main/java/com/lpvs/service/scanner/scanoss/LPVSScanossDetectService.java (hunk 4 above)
Comment (o-kopysov) [reply src/main/java/com/lpvs/service/scanner/scanoss/LPVSScanossDetectService.java:82]: Previous version of LPVSScanossDetectService is not testable at this point, which indicates the presence  of issues in architecture. That's why I have to make changes in the declaration of ProcessBuilder to make it mockable. LPVSScanossDetectService uses only scanoss scanner, so it doesn't depend on webhookconfig.

Regarding the update in GitHub actions, I make them to cover more lines and branches by tests, but they can probably be removed from current PR if needed.

File: .github/workflows/build.yml (hunk 2 above)
Comment (o-kopysov) [reply .github/workflows/build.yml]: Agree. I've removed this from PR.

File: .github/workflows/findbugs.yml (hunk 3 above)
Comment (o-kopysov) [reply .github/workflows/findbugs.yml]: Agree. I've removed this from PR.

**Comment by o-kopysov** [comment]
**JaCoCo Test Coverage Summary**

- Coverage: 92.846%
- Branches: 80.078%

**Review by tiokim — APPROVED** [review]
LGTM

**Pull request merged**